pkg/catalog/types.go

```diff
@@ -70,9 +70,13 @@ type MeshCataloger interface {
 	// ListSMIPolicies lists SMI policies.
 	ListSMIPolicies() ([]*split.TrafficSplit, []service.WeightedService, []service.K8sServiceAccount, []*spec.HTTPRouteGroup, []*target.TrafficTarget, []*corev1.Service)
 
-	// ListEndpointsForService returns the list of provider endpoints corresponding to a service
+	// ListEndpointsForService returns the list of individual instance endpoint destinations that comprise the backed service
 	ListEndpointsForService(service.MeshService) ([]endpoint.Endpoint, error)
 
+	// GetServiceEndpoint returns the highest abstract set of endpoint destinations where the service is made available at.
+	// If no LB/virtual IPs are assigned to the service, GetServiceEndpoints will return ListEndpointsForService
+	GetServiceEndpoints(service.MeshService) ([]endpoint.Endpoint, error)
```

Comment:
This would clear the ambiguity with `ListEndpointsForService`, where we attempt to convey that `GetResolvableEndpointsForService` returns only endpoints that are resolvable using service FQDN.

```suggestion
	GetResolvableEndpointsForService(service.MeshService) ([]endpoint.Endpoint, error)
```

Reply:
+1, much better